src/ecs/rendering/systems.rs

```diff
@@ -38,6 +38,46 @@ impl<'a> System<'a> for RenderSystem {
     ) {
         use std::time::Duration;
 
+        // FIXME issue when running in debug mode:
+        //https://github.com/gfx-rs/gfx/blob/master/src/core/src/pso.rs#L279
+        //Fetch current window size for this rendersystem
+        //Compare with current target size
+        if let Some(cur_window_size) = self.renderer.window_size() {
+            //Window size changed
+            for (name, target) in self.pipe.targets.iter_mut() {
+                if cur_window_size != target.size() {
+                    if let Some(new_target) = self.renderer.regen_target() {
+                        //Replace target in specific stage by name
+                        for stage in self.pipe.stages.iter_mut() {
+                            if stage.get_target_name() == *name {
+                                stage.set_target(new_target.clone());
+                                // Update stage effects, which are used to draw on the buffers
+                                for pass in stage.passes.iter_mut() {
+                                    let effect = &mut pass.effect.data;
+                                    // Update Color Buffer
+                                    effect.out_colors.clear();
+                                    effect.out_colors.extend(new_target
+                                        .color_bufs()
+                                        .iter()
+                                        .map(|cb| cb.as_output.clone())
+                                    );
+                                    // Update Depth Stencil
+                                    effect.out_depth = new_target.depth_buf().map(|db| {
+                                        (db.as_output.clone(), (0, 0))
+                                    });
+                                }
+                            }
+                        }
+                        if *name == "" {
+                            //Means this is the main target reference of the pipeline
+                            self.renderer.set_main_target(target.clone());
```

Comment:
Here you place old target into `renderer.main_target` field. Were you intend to do that?